### PR TITLE
Fix/UI hang on stop#17

### LINKED ISSUE #17: BUG 🐞: UI hangs sometimes when stopping a container
**Describe the bug**
The UI hangs sometimes when stopping (pressing `s`) a container randomly. 

**Expected behavior**
There should be no lag when i hit `s` to stop.

**Desktop:**
 - OS:  platform independent 
 - gmd Version: 1.3

**Additional context**
This is because we run `ToggleStartStop` on the main go routine, so if this operation takes time it naturally hangs the UI. 

```
	if curItem != nil {
		containerId := curItem.(dockerRes).getId()
		err := m.dockerClient.ToggleStartStopContainer(containerId)

		if err != nil {
			m.activeDialog = teadialog.NewErrorDialog(err.Error(), m.width)
			m.showDialog = true
		}

	}

```

So, the fix would be to run `ToggleStartStop` on a different go routine and use `m.possibleLongRunningOpErrorChan` channel to send errors if any. 

Let me know if any one needs any additional info.

### PR DESCRIPTION
**Pull request by ashishkhuraishy**
fixes #17

### CODE REVIEW

**Comment by ajayd-san** [comment]
Hey, great job on the PR!

Can you make `handleBackground` a `MainModel` method ? Everything else would be the same. We can use this method to port existing blocking operations in the next release.

Thanks

**Comment by ashishkhuraishy** [comment]
added it as part of the MainModel and refactored a couple of blocking functions

File: tui/mainModel.go [hunk 1]
```diff
@@ -851,6 +827,15 @@ func (m *MainModel) prepopulateContainerSizeMapConcurrently() {
...
 
+func (m MainModel) handleBackground(operation func(string) error) {
+	curItem := m.getSelectedItem()
+	if curItem == nil {
+		return
+	}
+	containerId := curItem.(dockerRes).getId()
+	m.possibleLongRunningOpErrorChan <- operation(containerId)
+}
+
```
Comment (ajayd-san) [inline tui/mainModel.go:838]: We should consider making this a pure function. 

i.e pass in `id` to the function as well. Cuz we cannot determine when a go routine will be scheduled to run. If the user somehow is selecting a different list item when `handleBackground` gets run, the `operation` will be run on that item instead. Which is not supposed to happen.
 
```suggestion
func (m MainModel) handleBackground(operation func(string) error, id string) {
	m.possibleLongRunningOpErrorChan <- operation(id)
}

```

File: tui/types.go
Comment (ajayd-san) [inline tui/types.go:1]: these changes are unrelated to this issue. I've already merged your PR #19  that has these changes.

File: tui/mainModel.go [hunk 2]
```diff
@@ -335,47 +340,22 @@ func (m MainModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
...
 				case key.Matches(msg, ContainerKeymap.ToggleListAll):
 					m.dockerClient.ToggleContainerListAll()
 
 				case key.Matches(msg, ContainerKeymap.ToggleStartStop):
 					log.Println("s pressed")
-					curItem := m.getSelectedItem()
-					if curItem != nil {
-						containerId := curItem.(dockerRes).getId()
-						err := m.dockerClient.ToggleStartStopContainer(containerId)
+					go m.handleBackground(m.dockerClient.ToggleStartStopContainer)
```
Comment (ajayd-san) [inline tui/mainModel.go:350]: and update these to get current Id and abide by the newly suggested function signature for `m.handleBackground`.

**Review by ajayd-san — CHANGES_REQUESTED** [review]
Hey, pl make the suggested changes. Thanks!

**Comment by ajayd-san** [comment]
Hey @ashishkhuraishy , If you are still working on this. I'd recommend you rebase against `notification_system`.

**Pull request closed without merging**